tidy3d/components/validators.py

```diff
@@ -85,19 +85,39 @@ def field_has_unique_names(cls, val, values):
 
 
 def assert_objects_in_sim_bounds(field_name: str):
-    """makes sure all objects in field are at least partially inside of simulation bounds/"""
+    """Makes sure all objects in field are at least partially inside of simulation bounds.
+    If a Mode object, this checks that the object is fully in the main quadrant in the presence
+    of symmetry along a given axis, or else centered on the symmetry center."""
 
     @pydantic.validator(field_name, allow_reuse=True, always=True)
     def objects_in_sim_bounds(cls, val, values):
         """check for intersection of each structure with simulation bounds."""
-        sim_bounds = Box(size=values.get("size"), center=values.get("center"))
+        sim_center = values.get("center")
+        sim_box = Box(size=values.get("size"), center=sim_center)
+
         for position_index, geometric_object in enumerate(val):
-            if not sim_bounds.intersects(geometric_object.geometry):
+            if not sim_box.intersects(geometric_object.geometry):
                 raise SetupError(
                     f"'{geometric_object}' "
-                    f"(at `simulation.{field_name}[{position_index}]`)"
-                    "is completely outside of simulation domain"
+                    f"(at `simulation.{field_name}[{position_index}]`) "
+                    "is completely outside of simulation domain."
                 )
+
+            if geometric_object.type in ["ModeSource", "ModeMonitor"]:
```

Comment:
seems like this should be a separate validator?

Reply:
I guess it can be yeah